File: credit.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

long x;
int s1;
int card(int a);
int d(long b);

int main(void)
{
    do
    {
        x = get_long("card number:");
    }
    while (x < 1);
    s1 = 0;
    for (int i = 2; i < 17; i = i + 2)
    {
        s1 = s1 + d(2 * card(i));
    }
    int s2 = 0;
    for (int j = 1; j < 17; j = j + 2)
    {
        s2 = card(j) + s2;
    }
    if ((s1 + s2) % 10 != 0)
    {
        printf("INVALID\n");
    }
    else if (card(16) == 4 || (card(13) == 4 && card(16) == 0 && card(15) == 0))
    {
        printf("VISA\n");
    }
    else if (card(16) == 5 &&
             (card(15) == 1 || card(15) == 2 || card(15) == 3 || card(15) == 4 || card(15) == 5))
    {
        printf("MASTERCARD\n");
    }
    else if (card(16) == 0 && (card(15) == 3 && (card(14) == 4 || card(14) == 7)))
    {
        printf("AMEX\n");
    }
    else
    {
        printf("INVALID\n");
    }
}
int card(int a)
{
    long y = x / pow(10, a - 1);
    return y % 10;
}
int d(long b)
{
    int g = 0;
    while (b != 0)
    {
        g = g + b % 10;
        b = b / 10;
    }
    return g;
}

File: credit.py
import cs50
def checksum(no):
    def convertlist(no):
        return [int(d) for d in str(no)]
    digits = convertlist(no)
    every_other = digits[-2::-2]
    therest = digits[-1::-2]
    summ = 0
    summ += sum(therest)
    for d in every_other:
        summ += sum(convertlist(d * 2))
    return summ % 10

while True:
    cardno = cs50.get_int("Card Number: ")
    if cardno < 1:
        print("Provide A Proper Card Number With Only Digits")
    else:
        break

if checksum(cardno) == 0:
    cardno = str(cardno)
    if len(cardno) == 15 and (int(cardno[0:2]) in [34, 37]):
        print("AMEX")

    elif len(cardno) == 16 and (int(cardno[0:2]) in [51, 52, 53, 54, 55]):
        print("MASTERCARD")

    elif len(cardno) in [13, 16] and int(cardno[0]) == 4:
        print("VISA")

    else:
        print("INVALID")
else:
    print("INVALID")

